src/python/pants/engine/rules.py

```diff
@@ -335,30 +345,32 @@ def rule(*args, cacheable=True) -> Callable:
                          f'Element {index} (0-based) {parameter_type} is of type '
                          f'{type(parameter_type)}')
     return _make_rule(return_type, parameter_types, cacheable=cacheable)
-  elif len(args) == 1 and inspect.isfunction(args[0]):
-    func = args[0]
-    signature = inspect.signature(func)
-    func_id = f'@rule {func.__module__}:{func.__name__}'
-    return_type = _ensure_type_annotation(
-      annotation=signature.return_annotation,
-      name=f'{func_id} return',
-      empty_value=inspect.Signature.empty,
-      raise_type=MissingReturnTypeAnnotation
+
+  if len(args) != 1 and not inspect.isfunction(args[0]):
```

Comment:
I know you like this style. In this case though your thrashing code just laid down. Did I violate a style guide? If so - all good. If not, seems worth establishing this there or thinking 2x before doing it.

Reply:
No, looks like our style guide is silent on this style of early returns / inverted conditionals: https://www.pantsbuild.org/styleguide.html

Noted for the future about not whimsically changing it. Lmk if you'd like me to change back here - happy to.